GVRf/Framework/framework/src/main/java/org/gearvrf/io/GVRMouseDeviceManager.java

```diff
@@ -75,9 +80,17 @@ void removeCursorController(GVRBaseController controller) {
             thread.quitSafely();
             thread = new EventHandlerThread(THREAD_NAME);
             threadStarted = false;
+            if(gvrContext != null){
+                gvrContext.unregisterDrawFrameListener(this);
+            }
         }
     }
 
+    @Override
+    public void onDrawFrame(float frameTime) {
+        thread.invalidate();
```

Comment:
Does this all have to be on the GL thread? There might be a small portion that has to be but most probably doesn't have. The user callbacks ideally should not be on the GL thread. The so called framework main thread seems a good candidate.